Cbindgen emits header file that can not be used with C

Because cbindgen emits a header such as:

```
#include <cstdarg>
#include <cstdint>
#include <cstdlib>
#include <new>

extern "C" {

uintptr_t do_nothing_rust();

} // extern "C"
```

This fails to compile when using C, because the 4 c++ headers can't be found:

```
In file included from /home/william/development/389ds/ds/ldap/servers/slapd/configdse.c:27:
/home/william/development/389ds/ds/ldap/servers/slapd/rust-slapi-private.h:1:10: fatal error: cstdarg: No such file or directory
    1 | #include <cstdarg>
      |          ^~~~~~~~~
```

These should be masked behind an ifdef for C++ only, so that they don't cause issues with C building. 

IE the emitted header should be:

```
#ifdef __cplusplus
#include <cstdarg>
#include <cstdint>
#include <cstdlib>
#include <new>

extern "C" {
#endif 

uintptr_t do_nothing_rust(); 

#ifdef __cplusplus
}
#endif   
```

--- a/README.md
+++ b/README.md
@@ -50,6 +50,8 @@ Then all you need to do is run it:
 cbindgen --config cbindgen.toml --crate my_rust_library --output my_header.h
 ```
 
+This produces a header file for C++.  For C, add the `--lang c` switch.
+
 See `cbindgen --help` for more options.
 
 [Read the full user docs here!](docs.md)

--- a/docs.md
+++ b/docs.md
@@ -36,6 +36,8 @@ Then all you need to do is run it:
 cbindgen --config cbindgen.toml --crate my_rust_library --output my_header.h
 ```
 
+This produces a header file for C++.  For C, add the `--lang c` switch.
+
 See `cbindgen --help` for more options.
 
 [Get a template cbindgen.toml here.](template.toml)

